### ISSUE TITLE
Black playback screen on Elementary OS 0.4/Loki (Ubuntu 16.04) & MLT 6.0.0.2; works in Shotcut

### ISSUE THREAD

**Issue by amsheppard** [jliljebl/flowblade#358]
Hello - Flowblade 1.12.0 only gives a black screen when tying to play back an imported .mp4 (1024x576 16:9 24fps). Audio plays back & the audio meters are responsive.

Shotcut version 17.03.02 (Shotcut.app) works as expected.

Skill level: I'm comfortable around the CLI... this video thing, though, I'm a newbie. Any suggestions?

# Environment
Kernel: Linux wrkstn00 4.4.0-62-lowlatency #83-Ubuntu SMP PREEMPT Wed Jan 18 15:00:41 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
Distro: Elementry OS 0.4/Loki (Ubuntu 16.04 based)
MLT framework: python-mlt 6.0.0-2 amd64

```
/usr/bin/python3 --version
Python 3.6.1
```
I have both python 3.5 & 3.6 installed.

## Video Controller
```
sudo lshw -c video
  *-display               
       description: VGA compatible controller
       product: Caicos [Radeon HD 6450/7450/8450 / R5 230 OEM]
       vendor: Advanced Micro Devices, Inc. [AMD/ATI]
       physical id: 0
       bus info: pci@0000:0c:00.0
       version: 00
       width: 64 bits
       clock: 33MHz
       capabilities: pm pciexpress msi vga_controller bus_master cap_list rom
       configuration: driver=radeon latency=0
       resources: irq:31 memory:c0000000-cfffffff memory:fc2c0000-fc2dffff ioport:dc00(size=256) memory:fc100000-fc11ffff
```
# CLI Startup output
```
FLOWBLADE MOVIE EDITOR 1.12
---------------------------
Launch script dir: /usr/bin
Running from installation...
modules path: /usr/share/flowblade/Flowblade
MLT found, version: 6.0.0
OS: elementary OS 0.4 Loki
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
[GCC 5.4.0 20160609]
GTK+ version: 3.18.9
User dir: /home/null/.flowblade/
Locale: en_US
Translations at /usr/share/locale were not found, using program root directory translations.
Use OS locale language.
3600 1463
Small height: False
Small width: False
Detecting environment...
---
video_codecs:
  - a64multi
  - a64multi5
  - alias_pix
  - amv
  - apng
  - asv1
  - asv2
  - avrp
  - avui
  - ayuv
  - bmp
  - cinepak
  - cljr
  - dnxhd
  - dpx
  - dvvideo
  - ffv1
  - ffvhuff
  - flashsv
  - flashsv2
  - flv
  - gif
  - h261
  - h263
  - h263p
  - hap
  - huffyuv
  - jpeg2000
  - jpegls
  - ljpeg
  - mjpeg
  - mpeg1video
  - mpeg2video
  - mpeg4
  - msmpeg4v2
  - msmpeg4
  - msvideo1
  - pam
  - pbm
  - pcx
  - pgm
  - pgmyuv
  - png
  - ppm
  - prores
  - prores_aw
  - prores_ks
  - qtrle
  - r10k
  - r210
  - rawvideo
  - roqvideo
  - rv10
  - rv20
  - sgi
  - snow
  - sunrast
  - svq1
  - targa
  - tiff
  - utvideo
  - v210
  - v308
  - v408
  - v410
  - wmv1
  - wmv2
  - xbm
  - xface
  - xwd
  - y41p
  - yuv4
  - zlib
  - zmbv
  - libopenjpeg
  - libschroedinger
  - libtheora
  - libvpx
  - libvpx-vp9
  - libwebp
  - libx264
  - libx264rgb
  - libx265
  - libxvid
...
---
audio_codecs:
  - comfortnoise
  - s302m
  - aac
  - ac3
  - ac3_fixed
  - alac
  - dca
  - eac3
  - flac
  - g723_1
  - mp2
  - mp2fixed
  - nellymoser
  - real_144
  - sonic
  - sonicls
  - tta
  - vorbis
  - wavpack
  - wmav1
  - wmav2
  - pcm_alaw
  - pcm_f32be
  - pcm_f32le
  - pcm_f64be
  - pcm_f64le
  - pcm_mulaw
  - pcm_s8
  - pcm_s8_planar
  - pcm_s16be
  - pcm_s16be_planar
  - pcm_s16le
  - pcm_s16le_planar
  - pcm_s24be
  - pcm_s24daud
  - pcm_s24le
  - pcm_s24le_planar
  - pcm_s32be
  - pcm_s32le
  - pcm_s32le_planar
  - pcm_u8
  - pcm_u16be
  - pcm_u16le
  - pcm_u24be
  - pcm_u24le
  - pcm_u32be
  - pcm_u32le
  - roq_dpcm
  - adpcm_adx
  - g722
  - g726
  - adpcm_ima_qt
  - adpcm_ima_wav
  - adpcm_ms
  - adpcm_swf
  - adpcm_yamaha
  - libgsm
  - libgsm_ms
  - libmp3lame
  - libopencore_amrnb
  - libopus
  - libshine
  - libspeex
  - libtwolame
  - libvo_aacenc
  - libvo_amrwbenc
  - libvorbis
  - libwavpack
...
---
formats:
  - a64
  - ac3
  - adts
  - adx
  - aiff
  - amr
  - apng
  - asf
  - ass
  - ast
  - asf_stream
  - au
  - avi
  - avm2
  - bit
  - caf
  - cavsvideo
  - crc
  - dash
  - data
  - daud
  - dirac
  - dnxhd
  - dts
  - dv
  - eac3
  - f4v
  - ffm
  - ffmetadata
  - filmstrip
  - flac
  - flv
  - framecrc
  - framemd5
  - g722
  - g723_1
  - gif
  - gxf
  - h261
  - h263
  - h264
  - hds
  - hevc
  - hls
  - ico
  - ilbc
  - image2
  - image2pipe
  - ipod
  - ircam
  - ismv
  - ivf
  - jacosub
  - latm
  - lrc
  - m4v
  - md5
  - matroska
  - matroska
  - microdvd
  - mjpeg
  - mlp
  - mmf
  - mov
  - mp2
  - mp3
  - mp4
  - mpeg
  - vcd
  - mpeg1video
  - dvd
  - svcd
  - mpeg2video
  - vob
  - mpegts
  - mpjpeg
  - mxf
  - mxf_d10
  - mxf_opatom
  - null
  - nut
  - oga
  - ogg
  - oma
  - opus
  - alaw
  - mulaw
  - f64be
  - f64le
  - f32be
  - f32le
  - s32be
  - s32le
  - s24be
  - s24le
  - s16be
  - s16le
  - s8
  - u32be
  - u32le
  - u24be
  - u24le
  - u16be
  - u16le
  - u8
  - psp
  - rawvideo
  - rm
  - roq
  - rso
  - rtp
  - rtp_mpegts
  - rtsp
  - sap
  - segment
  - stream_segment,ssegment
  - singlejpeg
  - smjpeg
  - smoothstreaming
  - sox
  - spx
  - spdif
  - srt
  - swf
  - tee
  - 3g2
  - 3gp
  - mkvtimestamp_v2
  - truehd
  - uncodedframecrc
  - vc1
  - vc1test
  - voc
  - w64
  - wav
  - webm
  - webm_dash_manifest
  - webm_chunk
  - webp
  - webvtt
  - wtv
  - wv
  - yuv4mpegpipe
  - alsa
  - caca
  - fbdev
  - opengl
  - oss
  - pulse
  - sdl
  - v4l2
  - xv
...
MLT detection succeeded, 152 formats, 84 video codecs and 68 audio codecs found.
388 MLT services found.
Loading render profiles...
Loading filters...
Brightness dropped, MLT version too low for this filter.
MLT service avfilter.colorchannelmixer not found.
MLT service avfilter.lut3d not found.
MLT service avfilter.lenscorrection not found.
MLT service avfilter.perspective not found.
MLT service avfilter.zoompan not found.
Loading transitions...
RGB Adjustment dropped for Color Adjustment
Hue dropped for Color Adjustment
Gamma dropped for Lift Gain Gamma
G'MIC found
Natron not found
Player initialized with profile:  HD 1080p 30 fps
Selected color detected
BG color detected
Autosave started...
Save project autosave_d27998e43d33cd21981b01353dead9dc
```

**Comment by jliljebl** [comment]
This is probably related tor Issues https://github.com/jliljebl/flowblade/issues/228 and https://github.com/jliljebl/flowblade/issues/309

There are indications that this might go away with xorg-server-1.19.0, which xorg you have?

Flowblade is Python2 application, btw.

**Comment by jliljebl** [comment]
Shotcut uses OpenGl to display video, Flowblade uses SDL, so this is "SDL video on your system" related.

**Comment by amsheppard** [comment]
Wow. That was quick. Thanks!

I saw that other issue on eOS 14.04 but saw no further followup. 

Is this what you mean?
```
dpkg -l | grep "sdl"
ii  libsdl-image1.2:amd64                          1.2.12-5build2                                            amd64        Image loading library for Simple DirectMedia Layer 1.2, libraries
ii  libsdl1.2debian:amd64                          1.2.15+dfsg1-3                                            amd64        Simple DirectMedia Layer
``

Here are my X.Org packages
```
ii  python3-xkit                                   0.5.0ubuntu2                                              all          library for the manipulation of xorg.conf files (Python 3)
ii  xorg                                           1:7.7+13ubuntu3                                           amd64        X.Org X Window System
ii  xorg-docs-core                                 1:1.7.1-1ubuntu1                                          all          Core documentation for the X.org X Window System
ii  xserver-xorg                                   1:7.7+13ubuntu3                                           amd64        X.Org X server
ii  xserver-xorg-core                              2:1.18.4-0ubuntu0.2                                       amd64        Xorg X server - core server
ii  xserver-xorg-input-all                         1:7.7+13ubuntu3                                           amd64        X.Org X server -- input driver metapackage
ii  xserver-xorg-input-evdev                       1:2.10.1-1ubuntu2                                         amd64        X.Org X server -- evdev input driver
ii  xserver-xorg-input-libinput                    0.18.0-1                                                  amd64        X.Org X server -- libinput input driver
ii  xserver-xorg-input-synaptics                   1.8.2-1ubuntu3                                            amd64        Synaptics TouchPad driver for X.Org server
ii  xserver-xorg-input-vmmouse                     1:13.1.0-1ubuntu2                                         amd64        X.Org X server -- VMMouse input driver to use with VMWare
ii  xserver-xorg-input-wacom                       1:0.32.0-0ubuntu3                                         amd64        X.Org X server -- Wacom input driver
ii  xserver-xorg-video-all                         1:7.7+13ubuntu3                                           amd64        X.Org X server -- output driver metapackage
ii  xserver-xorg-video-amdgpu                      1.1.2-0ubuntu0.16.04.1                                    amd64        X.Org X server -- AMDGPU display driver
ii  xserver-xorg-video-ati                         1:7.7.0-1                                                 amd64        X.Org X server -- AMD/ATI display driver wrapper
ii  xserver-xorg-video-fbdev                       1:0.4.4-1build5                                           amd64        X.Org X server -- fbdev display driver
ii  xserver-xorg-video-intel                       2:2.99.917+git20160706-1ubuntu1~elementary0.4.1           amd64        X.Org X server -- Intel i8xx, i9xx display driver
ii  xserver-xorg-video-nouveau                     1:1.0.12-1build2                                          amd64        X.Org X server -- Nouveau display driver
ii  xserver-xorg-video-qxl                         0.1.4-3ubuntu3                                            amd64        X.Org X server -- QXL display driver
ii  xserver-xorg-video-radeon                      1:7.7.0-1                                                 amd64        X.Org X server -- AMD/ATI Radeon display driver
ii  xserver-xorg-video-vesa                        1:2.3.4-1build2                                           amd64        X.Org X server -- VESA display driver
ii  xserver-xorg-video-vmware                      1:13.1.0-2ubuntu3                                         amd64        X.Org X server -- VMware display driver
```

**Comment by jliljebl** [comment]
Yeah, these are the related libraries.

if you give in terminal:

xgamma -gamma 1
flowblade

...does this help?

**Comment by amsheppard** [comment]
No luck. I trust I've not omitted anyting irrelevant:

```
MLT detection succeeded, 152 formats, 84 video codecs and 68 audio codecs found.
388 MLT services found.
Loading render profiles...
Loading filters...
Brightness dropped, MLT version too low for this filter.
MLT service avfilter.colorchannelmixer not found.
MLT service avfilter.lut3d not found.
MLT service avfilter.lenscorrection not found.
MLT service avfilter.perspective not found.
MLT service avfilter.zoompan not found.
Loading transitions...
RGB Adjustment dropped for Color Adjustment
Hue dropped for Color Adjustment
Gamma dropped for Lift Gain Gamma
G'MIC found
Natron not found
Player initialized with profile:  HD 1080p 30 fps
Selected color detected
BG color detected
Autosave started...
Save project autosave_ce8142520ce75f265e3144189ec2f6e0
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x7f754001bba0] stream 0, timescale not set
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x7f75404b6720] stream 0, timescale not set
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x7f754047dd80] stream 0, timescale not set
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x7f75401d97a0] stream 0, timescale not set
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x7f754001c580] stream 0, timescale not set
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x7f754048c2a0] stream 0, timescale not set
Save project untitled.flb
Saving changed profile project:  1024x576 16:9 24fps
FPS conversion multiplier: 0.8
Loading Project, SAVEFILE_VERSION: 5
Player initialized with profile:  1024x576 16:9 24fps
Autosave started...
Save project autosave_ce8142520ce75f265e3144189ec2f6e0
Updating media lengths:
Fantastic Four - Rise of the Silver Surfer - Trailer.mp4
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x7f750807d480] stream 0, timescale not set
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x7f750851d6e0] stream 0, timescale not set
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x7f7508502dc0] stream 0, timescale not set
Updating media lengths done.
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x38b36e0] stream 0, timescale not set
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x373b8a0] stream 0, timescale not set
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x307d400] stream 0, timescale not set
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x7f7538022500] stream 0, timescale not set
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x7f75380d0f20] stream 0, timescale not set
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x7f75380379a0] stream 0, timescale not set

(flowblade:25691): Gtk-CRITICAL **: gtk_combo_box_text_insert: assertion 'text != NULL' failed
Save project ff4.flb
Autosave started...
```

**Comment by jliljebl** [comment]
Ok. ElementaryOS on VirtualBox does not reproduce this, and no other distro has reported this, so I'm quite convinced that the problem is in ElementaryOS.

I'm closing and merging this with the earlier Issue, and change the title as it is not only NVidia related.

**Comment by amsheppard** [comment]
Sorry, I'm a touched confused. It _isn't_ reproducible in a eOs 0.4 vbox? I have a laptop w/ the same distro,  but hasn't been updated in a few weeks. It's also running the lowlatency kernel (to mirror my primary setup). Would you like me to try give it a go on that?

**Comment by jliljebl** [comment]
With host system Ubuntu 16.04 running Virtualbox  5.0.32_Ubuntu r112930 and on guest system elementaryOS Luna Flowblade displays video. It is unusably slow, but video is displayed.

You can try but the performance may be bad. Also if the host system is ElementaryOS problem may still occur, I'm not sure how video on VirtualBox works.

**Comment by amsheppard** [comment]
IDK what's up in that regard. I just installed Flowblade on my Thinkpad (after sorting out the dependencies). This is a 'bare metal' eOS 0.4, trying to play back sintel.mp4, same issue:

```
Xorg -version

X.Org X Server 1.18.4
Release Date: 2016-07-19
X Protocol Version 11, Revision 0
Build Operating System: Linux 4.4.0-45-generic x86_64 Ubuntu
Current Operating System: Linux x220-eos 4.4.0-59-lowlatency #80-Ubuntu SMP PREEMPT Fri Jan 6 21:04:05 UTC 2017 x86_64
Kernel command line: BOOT_IMAGE=/boot/vmlinuz-4.4.0-59-lowlatency root=UUID=b0dea471-4fd8-45c9-9444-6095db5dbdd6 ro quiet splash vt.handoff=7
Build Date: 02 November 2016  10:06:10PM
xorg-server 2:1.18.4-0ubuntu0.2 (For technical support please see http://www.ubuntu.com/support) 
Current version of pixman: 0.33.6
	Before reporting problems, check http://wiki.x.org
	to make sure that you have the latest version.
NotNull@x220-eos:~/Downloads$ dpkg -l | grep "sdl"
ii  libsdl-image1.2:amd64                          1.2.12-5build2                                          amd64        Image loading library for Simple DirectMedia Layer 1.2, libraries
ii  libsdl1.2debian:amd64                          1.2.15+dfsg1-3                                          amd64        Simple DirectMedia Layer
NotNull@x220-eos:~/Downloads$ sudo lshw -c video
  *-display               
       description: VGA compatible controller
       product: 2nd Generation Core Processor Family Integrated Graphics Controller
       vendor: Intel Corporation
       physical id: 2
       bus info: pci@0000:00:02.0
       version: 09
       width: 64 bits
       clock: 33MHz
       capabilities: msi pm vga_controller bus_master cap_list rom
       configuration: driver=i915 latency=0
       resources: irq:28 memory:f0000000-f03fffff memory:e0000000-efffffff ioport:5000(size=64)

FLOWBLADE MOVIE EDITOR 1.12
---------------------------
Launch script dir: /usr/bin
Running from installation...
modules path: /usr/share/flowblade/Flowblade
MLT found, version: 6.0.0
[...]
```

**Comment by jliljebl** [comment]
So Intel  Integrated video does not work wither. This "SDL video on elementary" issue almost certainly.

In my view it is highly unlikely that any modification to Flowblade or MLT source code would fix this, I see a few ways forward:
- reporting this as bug in elementaryOS and see if they are interested in taking action
- installing xorg-server-1.19.0 to see if that helps

**Comment by amsheppard** [comment]
HAH! I tried using the repo fr ```ppa:xorg-edgers/ppa``` to update XOrg. _That_ was a mistake! It almost hosed my install. Good thing I was able to ssh in to ```ppa-purge ppa:xorg-edgers/ppa && apt-get update && apt-get upgrade && apt-get install --reinstall elementary-desktop```

I'll shoot a ticket over to the eOS dev team & see what they can tell me.

Suggestion: perhaps link on the main description regarding this issue?

**Comment by jliljebl** [comment]
Yeah, updating xorg can destabilize the whole system.

I'm not going to put mention of a single distro issue on project front page.

**Comment by kodmgen** [comment]
You found solution?

**Comment by jliljebl** [comment]
This has good change for being fixed when SDL 1.2 consumer is replaced with some thing new and this is the main development focus right now as SDL1.2 is being dropped for SDL2 in MLT.
